### PR TITLE
Handling custom item links when merging contacts

### PR DESCRIPTION
**Pull request by escopecz**
| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [y]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes https://acquia.atlassian.net/browse/MAUT-11209

#### Description:

If 2 contacts are merged into 1 then the links to custom items from the loser contacts were lost. This PR adds them to the victor contacts instead.

#### Steps to test this PR:

1. Create 2 contacts
2. Create at least one custom object
3. Create at least one custom item
4. Link the custom item with first contact
5. Merge the first contact into the second one
6. Check that the second contact have the link from the loser contact.

### CODE REVIEW

File: EventListener/ContactSubscriber.php [hunk 1]
```diff
@@ -43,18 +46,25 @@ class ContactSubscriber implements EventSubscriberInterface
...
+    private $customItemXrefContactRepository;
+
     public function __construct(
         EntityManager $entityManager,
         TranslatorInterface $translator,
         CustomItemRouteProvider $routeProvider,
         CustomItemModel $customItemModel,
-        ConfigProvider $configProvider
+        ConfigProvider $configProvider,
+        CustomItemXrefContactRepository $customItemXrefContactRepository
```
Comment (tejasync) [inline EventListener/ContactSubscriber.php:60]: The entity manager is already injected, do you think there is a need for repo to be injected as well?

Comment (escopecz) [reply EventListener/ContactSubscriber.php:60]: @ts-navghane as we discussed during the team sync, DI is better than getting the repo from the entity manager, so let's keep it as is.

**Review by tejasync — APPROVED** [review]
Thanks @escopecz!

**Review by avikarshasaha — APPROVED** [review]
Tested in beta, it is working as described.
Thanks @escopecz

**Pull request merged**